### PR TITLE
Optimized battle gender checks

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Aka: Edu's PR Idea N°1 for when he had to rest his hands.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -11005,3 +10994,16 @@ static bool8 CanBeInfinitelyConfused(u8 battlerId)
     return TRUE;
 }
 
+u8 GetBattlerGender(u8 battlerId)
+{
+    return GetGenderFromSpeciesAndPersonality(gBattleMons[battlerId].species,
+                                              gBattleMons[battlerId].personality);
+}
+
+u8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2)
```
Comment (AgustinGDLV) [inline src/battle_util.c:11003]:
```suggestion
bool8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2)
```

File: include/battle_util.h [hunk 2]
```diff
@@ -235,5 +235,7 @@ u32 CountBattlerStatIncreases(u8 battlerId, bool32 countEvasionAcc);
 bool32 IsMyceliumMightOnField(void);
 bool8 ChangeTypeBasedOnTerrain(u8 battlerId);
 void RemoveConfusionStatus(u8 battlerId);
+u8 GetBattlerGender(u8 battlerId);
+u8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2);
```
Comment (AgustinGDLV) [inline include/battle_util.h:239]:
```suggestion
bool8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2);
```

**Review by AgustinGDLV — CHANGES_REQUESTED** [review]
I can't really find anything I'd want to change other than `u8` to `bool8` for `AreBattlersOfSameGender`. Really nice change, makes everything way more readable!

**Comment by AsparagusEduardo** [comment]
> `AreBattlersOfSameGender`

By this mean that I should rename `AreBattlersOfOppositeGender`?

**Comment by AgustinGDLV** [comment]
> > `AreBattlersOfSameGender`
> 
> By this mean that I should rename `AreBattlersOfOppositeGender`?

No I just can't read.

**Review by AgustinGDLV — APPROVED** [review]

**Pull request merged**